File: frontend/src/services/listeningSentenceService.js
import axiosClient from "../api/axiosClient";

const listeningSentenceService = {
  getByLesson(lessonId) {
    return axiosClient.get(`/v1/listening-sentences/lesson/${lessonId}`);
  },

  getById(id) {
    return axiosClient.get(`/v1/listening-sentences/${id}`);
  },

  create(data) {
    return axiosClient.post("/v1/listening-sentences", data);
  },

  update(id, data) {
    return axiosClient.put(`/v1/listening-sentences/${id}`, data);
  },

  delete(id) {
    return axiosClient.delete(`/v1/listening-sentences/${id}`);
  },

  reorder(lessonId, sentenceIds) {
    return axiosClient.post(
      `/v1/listening-sentences/reorder/${lessonId}`,
      sentenceIds,
    );
  },
};

export default listeningSentenceService;

File: backend/src/main/java/com/englishlearning/backend/service/ListeningSentenceService.java
package com.englishlearning.backend.service;

import com.englishlearning.backend.dto.request.ListeningSentenceRequest;
import com.englishlearning.backend.dto.response.ListeningSentenceResponse;

import java.util.List;

public interface ListeningSentenceService {
    ListeningSentenceResponse create(ListeningSentenceRequest request);
    ListeningSentenceResponse update(Long sentenceId, ListeningSentenceRequest request);
    void delete(Long sentenceId);
    ListeningSentenceResponse getById(Long sentenceId);
    List<ListeningSentenceResponse> getByLessonId(Long lessonId);
    List<ListeningSentenceResponse> reorderSentences(Long lessonId, List<Long> sentenceIds);
}